corehq/apps/dump_reload/management/commands/dump_case_data.py

```diff
@@ -0,0 +1,83 @@
+        if for_case_type:
+            available_case_types = (CommCareCaseSQL.objects.using(for_db_conn).
+                                    values_list('type', flat=True).distinct())
+            if for_case_type not in available_case_types:
+                raise CommandError("Unexpected case type {for_case_type} passed. "
+                                   "Should be from the following: {case_types}"
+                                   .format(for_case_type=for_case_type, case_types=available_case_types))
+
+        if for_domain:
+            available_domains = (CommCareCaseSQL.objects.using(for_db_conn).
+                                 values_list('domain', flat=True).distinct())
+            if for_domain not in available_domains:
+                raise CommandError("Domain name {for_domain} not found for any case in db {for_db_conn}".format(
+                    for_domain=for_domain, for_db_conn=for_db_conn
+                ))
+
+    def handle(self, *args, **options):
+        for_domain = options.get('domain')
+        for_case_type = options.get('case_type')
+        for_db_conn = options.get('db_conn')
+
+        if for_db_conn:
+            self.ensure_params(for_domain, for_case_type, for_db_conn)
+            dbs = [for_db_conn]
+        else:
+            dbs = settings.DATABASES
+
+        where_clause = "WHERE "
+        if for_domain:
+            where_clause += "domain='{domain}' and ".format(domain=for_domain)
+        for db in dbs:
+            db_conn = connections[db]
+            if 'form_processor_commcarecasesql' in db_conn.introspection.table_names():
+                available_case_types = (CommCareCaseSQL.objects.using('default').
+                                        values_list('type', flat=True).distinct())
+                if for_case_type and for_case_type not in available_case_types:
+                    print("Ignoring Unexpected case type {for_case_type} passed."
+                          "Should be from the following: {case_types}"
+                          .format(for_case_type=for_case_type, case_types=available_case_types))
+                    continue
+                if for_case_type:
+                    case_types = [for_case_type]
+                else:
+                    case_types = available_case_types
+                for case_type in case_types:
+                    file_name = "{case_type}_{db_name}_{timestamp}.csv".format(
+                        case_type=case_type, db_name=db,
+                        timestamp=datetime.utcnow().strftime(DATETIME_FORMAT))
+                    with open(file_name, "w") as output:
+                        c = db_conn.cursor()
+                        where_clause += "type='{case_type}' ".format(case_type=case_type)
+                        copy_query = "copy (SELECT * FROM form_processor_commcarecasesql " \
+                                     "{where_clause})".format(case_type=case_type, where_clause=where_clause)
```

Comment:
Rather than doing string formatting here you should use `psycopg2.sql` which is safer: http://initd.org/psycopg/docs/sql.html#module-psycopg2.sql

Reply:
`case_type` doesn't appear in the format string here.